app/helpers/puppetclasses_and_environments_helper.rb

```diff
@@ -17,7 +17,7 @@ def import_proxy_select(hash)
 
   def import_proxy_links(hash, classes = nil)
     SmartProxy.with_features("Puppet").map do |proxy|
-      display_link_if_authorized(_("Import from %s") % proxy.name, hash.merge(:proxy => proxy), :class=>classes)
+      display_link_if_authorized(_("Import from %s") % proxy.name, hash.merge(:proxy => proxy), {:class=>classes, :"data-no-turbolink" => true})
```

Comment:
I think that might make sense; however, is there any way we can speed up the page load? It currently fires several hundred queries at the database, as well as querying the smart proxy - both of them candidates for improvements.